romeo/src/state.rs

```diff
@@ -272,10 +272,6 @@ impl State {
 				withdrawals,
 				..
 			} => {
-				if status == TransactionStatus::Rejected {
```

Comment:
Let's deal with the unhappy path and do not crash. Just store the transaction status in the request

Reply:
What about returning with an empty vector if this is the case?
What happens if 
1. 
```rust
if status == TransactionStatus::Rejected{
return vec![];
}
```
2. Keep the other filtering changes.

Reply:
Then we loose the information that the stx tx failed and we will keep waiting for that tx. When we store the failed status the UI can also show that the withdrawal request failed.

Reply:
When we receive the stx tx status we should just update the state for this task.

Reply:
Crucial to resolve this for testnet - otherwise the current asset contract and all successful mints/burns can be made obsolete by one transaction that can't be fulfilled - as @friedger this could just be trying to withdraw more than the available balance.